### DISCUSSION TITLE
PERSONAL BIO PROJECT React: Promise pending on delete useEffect

### DISCUSSION THREAD

**Discussion by jimguistical** [nss-evening-cohort-14/evening-client-side#251] [Help Me!] [answered]
Jim Conner
Have you searched online and other tickets?  Yes

What have you tried? 
- I've tried refactoring code for handleClick (removed Switch) and reviewing deletePromise promise, and console warning to confirm handleClick working.  
![image](https://user-images.githubusercontent.com/76793747/119593440-96a86480-bd9f-11eb-88d4-01ae35ba151d.png)

https://github.com/jim-conner/personal-bio-react

https://github.com/jim-conner/personal-bio-react/tree/delete-projects
https://github.com/jim-conner/personal-bio-react/blob/delete-projects/src/App/components/ProjectCard.js
https://github.com/jim-conner/personal-bio-react/blob/delete-projects/src/helpers/data/projectData.js

**Comment by drteresavasquez** [comment]
What have you researched?

**Reply by jimguistical** [reply]
https://stackoverflow.com/questions/38401902/onclick-not-working-react-js
https://stackoverflow.com/questions/54873595/change-state-whenever-button-is-clicked-back-and-forth-in-react
https://stackoverflow.com/questions/61469748/react-component-doesnt-update-after-first-button-click
https://reactjs.org/docs/handling-events.html

https://stackoverflow.com/questions/60219144/unable-to-select-element-node-cannot-be-found
(also just got "Node cannot be found in the current page" while on projects but wasn't sure if it was related)

**Comment by TrinityChristiana** [comment]
It looks like you’re [console logging the promise itself](https://github.com/jim-conner/personal-bio-react/blob/c20ed8a813bf6406c6ea6e2b1b41965732327920/src/App/components/ProjectCard.js#L36) instead of the promises result. Remember because a promise is asynchronous, to get the result of it we have to `.then`  Try this:

```js
deleteProject(projectObj.firebaseKey).then((response) => console.log(response));
```

**Reply by jimguistical** [reply]
Thanks!  I tried updating my console warn, and I'm getting some odd behavior on the DOM -- tried to capture what I'm  [experiencing in a quick loom video](https://www.loom.com/share/1f664a9ac565402da0545d57ab6b695b)
At the end of the video when I refer to Projects, I'm referring to [this file ](https://github.com/jim-conner/personal-bio-react/blob/delete-projects/src/App/Views/Projects.js).  I am looking into how my useEffect for get Projects might be affecting my .map()

**Reply by jimguistical** [reply]
Reviewing #238 to see if my issue is related to Katy's re: useEffect and having an empty dependency array.

**Comment by jimguistical** [comment, marked as answer]
I (re)learned a valuable lesson: I needed to move useEffect out of where I was mapping over the project cards.  I don't think I was able to resolve the promise successfully with that useEffect in projects.js.
Then passing the whole obj instead of just projectObj.firebaseKey fixed my delete:
`deleteProject(projectObj)
        .then((projectsArray) => setProjects(projectsArray));`

**Comment by drteresavasquez** [comment]
Jim is this resolved?

**Reply by jimguistical** [reply]
Yep issue resolved!

**Reply by drteresavasquez** [reply]
Awesome!
